src/main.cpp

```diff
@@ -22,10 +22,23 @@
  *
 \******************************************************************************/
 
+#ifdef HEADLESS
+#    if defined( Q_OS_IOS )
+#        error HEADLES mode is not valid for IOS
+#    endif
+#    if defined( ANDROID )
+#        error HEADLES mode is not valid for ANDROID
+#    endif
+#endif
+
 #include <QCoreApplication>
 #include <QDir>
 #include <iostream>
 #include "global.h"
+#include "settings.h"
+#include "util.h"
+#include <memory>
```

Comment:
Did you just move code around here? Why?

Reply:
To keep things more clear....
All those ifdef's are confusing enough, so whenever possible let's keep common things together...

Reply:
> Better to avoid making unnecessary changes entirely.

In my opinion that is one of the main problems in the Jamulus development.
Just fixing bugs locally, without solving the real cause of the bug... often poor framework...
Improvements of the framework aren't "unnecessary changes"!
Real development is continuously improving the framework to prevent future similar bugs.
So you should strive for solutions that work in any case, not just for the bug you are fixing.